### ISSUE TITLE
Fill in the cholesky of the u0_prior in the rrule for the kalman

### ISSUE THREAD

**Issue by jlperla** [SciML/DifferenceEquations.jl#48]
Recall that a MvNormal distribution type used for the initial prior includes a vector for the mean and a PDMat for the covariance matrix.  In the custom sensitivity wrt that parameter we then need to fill in the Tangent type for the PDMat.

The tricky thing there is that a PDMat contains both the cholesky of the covariance matrix (in a `.chol` file, of the `Cholesky` type and basically implemented as `pd.chol = cholesky(pd.mat)` in a constructor) and its expanded product (in a `.mat` field).

In the Kalman filter, we are currently filling in the sensitivity for the `.mat` field but leaving the `.chol` as a NotImplemented.

I think the correct way to do this is to take the sensitivity for the `.mat` and then run it through the forward rule for the `cholesky`.  Something like
```julia
M = [1.0 0.1; 0.1 1.0]
p = PDMat(M)

# do stuff with p, using p.mat and p.chol
ΔD = # my internal calculation generating the change for p.mat

function mat_to_chol_sensitivity(ΔD, mat)
_, ΔC= frule((NoTangent(), ΔD), cholesky, mat) # note: rule does not currently exist
return ΔC

end
ΔC = @thunk(mat_to_chol_sensitivity(ΔD, mat)) # expensive calculation
Δp = Tangent{typeof(p)}(;mat=ΔD, chol = ΔC)
```
If that is correct-ish then the issue is that I there doesn't seem to be an frule defined for cholesky yet in ChainRules. The rule is in https://arxiv.org/pdf/1602.07527.pdf  if it is ever strictly necessary.

**Comment by wupeifan** [comment]
Yes, the rule in that pdf file is the forward one for cholesky decomposition. I'm not sure why it is frule instead of rrule for the setting above?
